kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieContainerInstanceImpl.java

```diff
@@ -49,11 +51,39 @@ public KieContainerInstanceImpl(String containerId, KieContainerStatus status, I
         super();
         this.kieContainer = kieContainer;
         this.resource = new KieContainerResource( containerId, null, status );
+        // set the default scanner state to DISPOSED (which is the actual default state)
+        // this way we don't need to do null checks all around for the scanner resource
```

Comment:
Here I am not 100% sure this is true. Is there any difference between `DISPOSED` and null scanner? Do we care if the scanner was ever set before (and thus is not null), and if it has not been touched at all (and thus is null)?